### PR TITLE
Fix Appveyor tests

### PR DESCRIPTION
**Pull request by arcanis**

### CODE REVIEW

**Comment by arcanis** [comment]
That should do the trick - the problem was that `getPossibleLocation` was splitting the path depending on path.sep, despite the OS accepting both backslashes and forward slashes as valid path (and the new test was using forward slashs).

I really think we should eventually try to normalize our implementation so that Yarn **always** use forward slashes inside the application code, leaving the utils implementation decide on what to do with these paths. There's some early work on this inside the fetchers PR.

File: src/registries/npm-registry.js [hunk 1]
```diff
@@ -110,7 +110,7 @@ export default class NpmRegistry extends Registry {
       [false, path.join(getGlobalPrefix(), filename)],
     ];
 
-    const foldersFromRootToCwd = this.cwd.split(path.sep);
+    const foldersFromRootToCwd = this.cwd.split(/[\\/]/g);
```
Comment (BYK) [inline src/registries/npm-registry.js]: How about doing
```
this.cwd = path.normalize(this.cwd);
const foldersFromRootToCwd = this.cwd.split(path.sep);
```

instead? See https://nodejs.org/api/path.html#path_path_normalize_path

**Comment by BYK** [comment]
Btw. thanks a lot for doing this! :)

File: src/registries/npm-registry.js (hunk 1 above)
Comment (cpojer) [inline src/registries/npm-registry.js]: There is also the "slashes" module on npm.

Comment (arcanis) [inline src/registries/npm-registry.js]: Normalize would work when using posix path on windows, but not the opposite (on posix, backslashes are valid filename characters, so path.posix doesn't convert them) :(

I'm gonna use `slash`, but I don't understand why it does an ASCII check ... filenames can definitely contain special characters, especially when dealing with user folders (oh how many times I've been burned when Windows put my home folder under `C:\Users\Maël` :().

**Pull request merged**